inc/namespace.php

```diff
@@ -503,6 +517,83 @@ function load_plugins() {
 	}
 }
 
+/**
+ * Ensure Tachyon URL is using the current site hostname.
+ *
+ * @param string $tachyon_url The current tachyon URL.
+ * @return string The updated Tachyon URL.
+ */
+function set_tachyon_hostname( string $tachyon_url ) : string {
+	$tachyon_host = wp_parse_url( $tachyon_url, PHP_URL_HOST );
+	$current_host = wp_parse_url( site_url(), PHP_URL_HOST );
+
+	if ( ! $tachyon_host ) {
+		trigger_error( sprintf( 'Error parsing Tachyon URL: %s', esc_url_raw( $tachyon_url ) ), E_USER_WARNING );
+		return $tachyon_url;
+	}
+
+	if ( ! $current_host ) {
+		trigger_error( sprintf( 'Error parsing current site URL: %s', esc_url_raw( site_url() ) ), E_USER_WARNING );
+		return $tachyon_url;
+	}
+
+	// Only do the replacement if the host name is not a subdomain of the Tachyon host.
+	if ( substr( $current_host, -strlen( $tachyon_host ) ) !== $tachyon_host ) {
```

Comment:
Can we make this `-1 * strlen ...` instead? It's clearer than an easily missed symbol and more apparent that you're applying a mathematical operator rather than a weird invocation of a function.
```suggestion
	if ( substr( $current_host, -1 * strlen( $tachyon_host ) ) !== $tachyon_host ) {
```